### PR TITLE
Fix outstanding reports spec

### PR DESCRIPTION
**Pull request by shalapatil**
## Summary
Test was flaky because of order of invoices. Now, fixed the outstanding reports code to return invoices in the descending order of updated_at. Fixed the test accordingly. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by apoorv1316 — APPROVED** [review]

File: app/models/client.rb [hunk 1]
```diff
@@ -115,7 +115,9 @@ def payment_summary(duration)
 
   def outstanding_and_overdue_invoices
```
Comment (prasanthchaduvula) [inline app/models/client.rb:116]: @shalapatil Please extract this logic to a reusable service object and use it in the controllers wherever required. As of now, it is needed for `InternalApi::V1::Reports::OutstandingOverdueInvoicesController`. Update the tests accordingly.

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/models/client.rb (hunk 1 above)
Comment (shalapatil) [reply app/models/client.rb:116]: can you please approve if made change is fine. Refactoring can be done in separate commit. This PR is to make sure other PR passes successfully.

**Comment by vipulnsward** [comment]
Specs failing, please assign only on green or mention if its a flaky spec

**Review by vipulnsward — APPROVED** [review]
Merge on green

**Pull request merged**